### PR TITLE
Router guard

### PR DESCRIPTION
**Pull request by Minghui-Gao**
user name can be changed as well
add profile to navigation bar
block access to the profile for users who are not logged in

### CODE REVIEW

File: backend/controllers/profileController.js [hunk 1]
```diff
@@ -24,12 +25,18 @@ const updateProfile = async (req, res) => {
...
+        await User.updateOne({ email },
+            {
+                name: req.body.nickName,
+            })
 
+        const currentProfile = await Profile.findOne({ email })
+        const currentUser = await User.findOne({ email })
 
         return res.status(200).send({
             msg: 'Update profile successfully',
```
Comment (LiamANeeson) [inline backend/controllers/profileController.js:37]: Could add a Toast notification for this maybe

Comment (LiamANeeson) [inline backend/controllers/profileController.js:37]: Could add a toast notification for this maybe

File: client/src/Components/RouterGuard/routerGuard.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+function AuthRoute({ children }) {
+    //check if login
+    const isLogin = localStorage.getItem('profile')
+
+    // If login, render directly and normally
+    if (isLogin) {
+        return <>{children}</>
+    }
+    // if not, Redirect to login route
```
Comment (LiamANeeson) [inline client/src/Components/RouterGuard/routerGuard.js:12]: Good Work here

File: client/src/Pages/Profile/UpdateProfile.js [hunk 3]
```diff
@@ -37,7 +38,8 @@ function UpdateProfile() {
         contactNumber: currentProfile.contactNumber,
         currentAddress: currentProfile.currentAddress,
```
[the remark this answers is not in the document]
Comment (Minghui-Gao) [reply client/src/Pages/Profile/UpdateProfile.js:39]: remove both contact number and current address or just current address?

File: backend/controllers/profileController.js (hunk 1 above)
Comment (Minghui-Gao) [reply backend/controllers/profileController.js:37]: I add this toast notification in the client, is that ok?

**Review by prajakta-kulkarni-git — APPROVED** [review]
looks good. I think we don't need this fields now.

**Pull request merged**